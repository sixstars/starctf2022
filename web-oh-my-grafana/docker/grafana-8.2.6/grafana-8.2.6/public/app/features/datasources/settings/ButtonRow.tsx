import React, { FC } from 'react';
import { selectors } from '@grafana/e2e-selectors';

import config from 'app/core/config';
import { Button, LinkButton } from '@grafana/ui';

import { AccessControlAction } from 'app/types/';
import { contextSrv } from 'app/core/core';

export interface Props {
  exploreUrl: string;
  isReadOnly: boolean;
  onDelete: () => void;
  onSubmit: (event: any) => void;
  onTest: (event: any) => void;
}

const ButtonRow: FC<Props> = ({ isReadOnly, onDelete, onSubmit, onTest, exploreUrl }) => {
  const canEditDataSources = !isReadOnly && contextSrv.hasPermission(AccessControlAction.DataSourcesWrite);
  const canDeleteDataSources = !isReadOnly && contextSrv.hasPermission(AccessControlAction.DataSourcesDelete);
  const canExploreDataSources = contextSrv.hasPermission(AccessControlAction.DataSourcesExplore);

  return (
    <div className="gf-form-button-row">
      <LinkButton variant="secondary" fill="solid" href={`${config.appSubUrl}/datasources`}>
        Back
      </LinkButton>
      <LinkButton variant="secondary" fill="solid" href={exploreUrl} disabled={!canExploreDataSources}>
        Explore
      </LinkButton>
      <Button
        type="button"
        variant="destructive"
        disabled={!canDeleteDataSources}
        onClick={onDelete}
        aria-label={selectors.pages.DataSource.delete}
      >
        Delete
      </Button>
      {canEditDataSources && (
        <Button
          type="submit"
          variant="primary"
          disabled={!canEditDataSources}
          onClick={(event) => onSubmit(event)}
          aria-label={selectors.pages.DataSource.saveAndTest}
        >
          Save &amp; test
        </Button>
      )}
      {!canEditDataSources && (
        <Button type="submit" variant="primary" onClick={onTest}>
          Test
        </Button>
      )}
    </div>
  );
};

export default ButtonRow;
