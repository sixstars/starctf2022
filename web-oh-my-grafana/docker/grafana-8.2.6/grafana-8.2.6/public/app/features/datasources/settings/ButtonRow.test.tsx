import React from 'react';
import { shallow } from 'enzyme';
import ButtonRow, { Props } from './ButtonRow';

jest.mock('app/core/core', () => {
  return {
    contextSrv: {
      hasPermission: () => true,
    },
  };
});

const setup = (propOverrides?: object) => {
  const props: Props = {
    isReadOnly: true,
    onSubmit: jest.fn(),
    onDelete: jest.fn(),
    onTest: jest.fn(),
    exploreUrl: '/explore',
  };

  Object.assign(props, propOverrides);

  return shallow(<ButtonRow {...props} />);
};

describe('Render', () => {
  it('should render component', () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with buttons enabled', () => {
    const wrapper = setup({
      isReadOnly: false,
    });

    expect(wrapper).toMatchSnapshot();
  });
});
