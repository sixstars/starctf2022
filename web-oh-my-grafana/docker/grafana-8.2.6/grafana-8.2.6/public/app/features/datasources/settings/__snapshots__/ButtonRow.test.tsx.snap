// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render should render component 1`] = `
<div
  className="gf-form-button-row"
>
  <LinkButton
    fill="solid"
    href="/datasources"
    variant="secondary"
  >
    Back
  </LinkButton>
  <LinkButton
    disabled={false}
    fill="solid"
    href="/explore"
    variant="secondary"
  >
    Explore
  </LinkButton>
  <Button
    aria-label="Data source settings page Delete button"
    disabled={true}
    onClick={[MockFunction]}
    type="button"
    variant="destructive"
  >
    Delete
  </Button>
  <Button
    onClick={[MockFunction]}
    type="submit"
    variant="primary"
  >
    Test
  </Button>
</div>
`;

exports[`Render should render with buttons enabled 1`] = `
<div
  className="gf-form-button-row"
>
  <LinkButton
    fill="solid"
    href="/datasources"
    variant="secondary"
  >
    Back
  </LinkButton>
  <LinkButton
    disabled={false}
    fill="solid"
    href="/explore"
    variant="secondary"
  >
    Explore
  </LinkButton>
  <Button
    aria-label="Data source settings page Delete button"
    disabled={false}
    onClick={[MockFunction]}
    type="button"
    variant="destructive"
  >
    Delete
  </Button>
  <Button
    aria-label="Data source settings page Save and Test button"
    disabled={false}
    onClick={[Function]}
    type="submit"
    variant="primary"
  >
    Save & test
  </Button>
</div>
`;
